### PR TITLE
refactor(atomic): second phase search box suggestions refactor

### PR DESCRIPTION
**Pull request by olamothe**
Part two/sister PR of https://github.com/coveo/ui-kit/pull/3787

Add back accessibility concerns (aria-messages) + re-enable commented out tests

KIT-3036

### CODE REVIEW

File: packages/atomic/src/components/insight/atomic-insight-search-box/atomic-insight-search-box.tsx [hunk 1]
```diff
@@ -294,15 +299,36 @@ export class AtomicInsightSearchBox {
...
     this.isExpanded = true;
-    this.suggestionManager.triggerSuggestions();
+    await this.suggestionManager.triggerSuggestions();
+    this.announceNewSuggestionsToScreenReader();
+  }
+
+  private announceNewActiveSuggestionToScreenReader() {
+    const ariaLabel = this.suggestionManager.activeDescendantElement?.ariaLabel;
+    if (isMacOS() && ariaLabel) {
+      this.suggestionsAriaMessage = ariaLabel;
```
Comment (fpbrault) [inline packages/atomic/src/components/insight/atomic-insight-search-box/atomic-insight-search-box.tsx:318]: What is this used for exactly?

**Review by fpbrault — APPROVED** [review]

File: packages/atomic/src/components/insight/atomic-insight-search-box/atomic-insight-search-box.tsx [hunk 2]
```diff
@@ -294,15 +299,36 @@ export class AtomicInsightSearchBox {
...
+    }
+  }
+
+  private announceNewSuggestionsToScreenReader() {
+    const elsLength =
```
Comment (louis-bompart) [inline packages/atomic/src/components/insight/atomic-insight-search-box/atomic-insight-search-box.tsx:323]: els?

**Review by louis-bompart — APPROVED** [review]
LGTM, unsure what an els is

File: packages/atomic/src/components/insight/atomic-insight-search-box/atomic-insight-search-box.tsx (hunk 2 above)
Comment (olamothe) [reply packages/atomic/src/components/insight/atomic-insight-search-box/atomic-insight-search-box.tsx:323]: That was old code I blindly reused / copy pasted; It's for "elementsLength"; I'll rename

**Pull request merged**